examples/sp/sp_tutorial.py

```diff
@@ -0,0 +1,316 @@
+# To learn more about the Spatial Pooler have a look at BAMI:
+# http://numenta.com/biological-and-machine-intelligence/
+# or at its class reference in the NuPIC documentation:
+# http://numenta.org/docs/nupic/classnupic_1_1research_1_1spatial__pooler_1_1_spatial_pooler.html
+# The purpose of the Spatial Pooler is to create a sparse representation
+# of its inputs in such a way that similar inputs will be mapped to similar
+# sparse representations. Thus, the Spatial Pooler should exhibit some resilience
+# to noise in its input.
+
+random.seed(1)
+uintType = "uint32"
+inputDimensions = (1000,1)
+columnDimensions = (2048,1)
+inputSize = np.array(inputDimensions).prod()
+columnNumber = np.array(columnDimensions).prod()
+inputArray = np.zeros(inputSize, dtype=uintType)
+
+for i in range(inputSize):
+  inputArray[i] = random.randrange(2)
+  
+activeCols = np.zeros(columnNumber, dtype=uintType)
+sp = SP(inputDimensions, 
+  columnDimensions, 
+  potentialRadius = int(0.5*inputSize),
+  numActiveColumnsPerInhArea = int(0.02*columnNumber),
+  globalInhibition = True,
+  seed = 1,
+  synPermActiveInc = 0.01,
+  synPermInactiveDec = 0.008
+   )
+
+print ""
+print "================================================="
+print "All images generated by this script will be saved"
+print "in your current working directory."
+print "================================================="
+print ""
+
+# Part 1: 
+# -------
+# A column connects to a subset of the input vector (specified
+# by both the potentialRadius and potentialPct). The overlap score
+# for a column is the number of connections to the input that become
+# active when presented with a vector. When learning is 'on' in the SP,
+# the active connections are reinforced, whereas those inactive are
+# depressed (according to parameters synPermActiveInc and synPermInactiveDec.
+# In order for the SP to create a sparse representation of the input, it
+# will select a small percentage (usually 2%) of its most active columns,
+# ie. columns with the largest overlap score.
+# In this first part, we will create a histogram showing the overlap scores
+# of the Spatial Pooler (SP) after feeding it with a random binary
+# input. As well, the histogram will show the scores of those columns
+# that are chosen to build the sparse representation of the input.
+
+sp.compute(inputArray, False, activeCols)
+overlaps = sp.getOverlaps()
+activeColsScores = []
+for i in activeCols.nonzero():
+  activeColsScores.append(overlaps[i])
+
+print ""
+print "---------------------------------"
+print "Figure 1 shows an histogram of the overlap scores"
+print "from all the columns in the spatial pooler, as well as the"
+print "overlap scores of those columns that were selected to build a"
+print "sparse representation of the input (shown in green)."
+print "The SP chooses 2% of the columns with the largest overlap score"
+print "to make such sparse representation."
+print "---------------------------------"
```

Comment:
@subutai Yes, it is the very first printout. Lines 115-120. Do you think they are not visible? I could print it out as the very last message of the program.